src/browser-detector/BrowserDetector.d.ts

```diff
@@ -0,0 +1,12 @@
+import { PureComponent } from 'react';
+
+type Browser = 'chrome' | 'firefox' | 'safari' | 'msie' | 'msedge' | 'opera';
+
+export type BrowserDetectorProps = {
+  children: any, // TODO: t.ReactChildren
+  placeholder: (detectedBrowser: string) => any, // TODO: t.ReactChildren
```

Comment:
Since `bowser` is defining its own types, it seems there's no way to use the ones from `@types/bowser` [](https://github.com/lancedikson/bowser/issues/182)
I cut&paste the relevant interface